app/src/app/allplayerstats/page.tsx

```diff
@@ -20,11 +20,13 @@ export default function AllPlayerStats() {
           .then(data => setStats(data))
           .catch(err => setError(err));
 
-        getBookmarks()
-            .then((data) => {
-                let pids = data.data.map((marked: any) => marked["pid"])
-                setBookmarks(pids)
-            });
+        if (session) {
+            getBookmarks()
+                .then((data) => {
+                    let pids = data.data.map((marked: any) => marked["pid"])
```

Comment:
due to construction of getBookmarks, data and data.data might be null, we should check these so there aren't errors and default to empty bookmarks if any errors, same with below